In [1]:
#number 1 question
A=[2,3,4,5] #I made up values to test if the '.extend()' method gives the required output
B=[3,4,5,7]
A_B=A.extend(B)
print(A_B)

None


In [2]:
A_B=A.append(B)
print(A_B)

None


In [3]:
#number 2 question
np.identity(3)

NameError: name 'np' is not defined

In [4]:
#this doesn't work
np.array[(1,0,0),(0,1,0),(0,0,1)]

NameError: name 'np' is not defined

In [5]:
#so does this
np.array([1,0,0],[0,1,0],[0,0,1])

NameError: name 'np' is not defined

In [6]:
#and this
eye(3)

NameError: name 'eye' is not defined

# FOR THE DATABASE

In [7]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

In [8]:
#importing data
data="https://raw.githubusercontent.com/WalePhenomenon/climate_change/master/fuel_ferc1.csv"
energy=pd.read_csv(data)
energy.head(5)

record_id  utility_id_ferc1  report_year  \
0   f1_fuel_1994_12_1_0_7                 1         1994   
1  f1_fuel_1994_12_1_0_10                 1         1994   
2   f1_fuel_1994_12_2_0_1                 2         1994   
3   f1_fuel_1994_12_2_0_7                 2         1994   
4  f1_fuel_1994_12_2_0_10                 2         1994   

       plant_name_ferc1 fuel_type_code_pudl fuel_unit  fuel_qty_burned  \
0              rockport                coal       ton        5377489.0   
1  rockport total plant                coal       ton       10486945.0   
2                gorgas                coal       ton        2978683.0   
3                 barry                coal       ton        3739484.0   
4             chickasaw                 gas       mcf          40533.0   

   fuel_mmbtu_per_unit  fuel_cost_per_unit_burned  \
0               16.590                      18.59   
1               16.592                      18.58   
2               24.130                      39.72   
3               23.950                      47.21   
4                1.000                       2.77   

   fuel_cost_per_unit_delivered  fuel_cost_per_mmbtu  
0                         18.53                1.121  
1                         18.53                1.120  
2                         38.12                1.650  
3                         45.99                1.970  
4                          2.77                2.570

In [9]:
#Cleaning the data
#To see all the information on the data
energy.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29523 entries, 0 to 29522
Data columns (total 11 columns):
record_id                       29523 non-null object
utility_id_ferc1                29523 non-null int64
report_year                     29523 non-null int64
plant_name_ferc1                29523 non-null object
fuel_type_code_pudl             29523 non-null object
fuel_unit                       29343 non-null object
fuel_qty_burned                 29523 non-null float64
fuel_mmbtu_per_unit             29523 non-null float64
fuel_cost_per_unit_burned       29523 non-null float64
fuel_cost_per_unit_delivered    29523 non-null float64
fuel_cost_per_mmbtu             29523 non-null float64
dtypes: float64(5), int64(2), object(4)
memory usage: 2.5+ MB


In [10]:
energy.describe(include='all')

record_id  utility_id_ferc1   report_year  \
count                      29523      29523.000000  29523.000000   
unique                     29523               NaN           NaN   
top     f1_fuel_2010_12_145_0_11               NaN           NaN   
freq                           1               NaN           NaN   
mean                         NaN        118.601836   2005.806050   
std                          NaN         74.178353      7.025483   
min                          NaN          1.000000   1994.000000   
25%                          NaN         55.000000   2000.000000   
50%                          NaN        122.000000   2006.000000   
75%                          NaN        176.000000   2012.000000   
max                          NaN        514.000000   2018.000000   

       plant_name_ferc1 fuel_type_code_pudl fuel_unit  fuel_qty_burned  \
count             29523               29523     29343     2.952300e+04   
unique             2315                   6         9              NaN   
top           big stone                 gas       mcf              NaN   
freq                156               11486     11354              NaN   
mean                NaN                 NaN       NaN     2.622119e+06   
std                 NaN                 NaN       NaN     9.118004e+06   
min                 NaN                 NaN       NaN     1.000000e+00   
25%                 NaN                 NaN       NaN     1.381700e+04   
50%                 NaN                 NaN       NaN     2.533220e+05   
75%                 NaN                 NaN       NaN     1.424034e+06   
max                 NaN                 NaN       NaN     5.558942e+08   

        fuel_mmbtu_per_unit  fuel_cost_per_unit_burned  \
count          29523.000000               29523.000000   
unique                  NaN                        NaN   
top                     NaN                        NaN   
freq                    NaN                        NaN   
mean               8.492111                 208.649031   
std               10.600220                2854.490090   
min                0.000001                -276.080000   
25%                1.024000                   5.207000   
50%                5.762694                  26.000000   
75%               17.006000                  47.113000   
max              341.260000              139358.000000   

        fuel_cost_per_unit_delivered  fuel_cost_per_mmbtu  
count                   2.952300e+04         29523.000000  
unique                           NaN                  NaN  
top                              NaN                  NaN  
freq                             NaN                  NaN  
mean                    9.175704e+02            19.304354  
std                     6.877593e+04          2091.540939  
min                    -8.749370e+02           -41.501000  
25%                     3.778500e+00             1.940000  
50%                     1.737100e+01             4.127000  
75%                     4.213700e+01             7.745000  
max                     7.964521e+06        359278.000000

In [11]:
#to see how many years are reported
energy.report_year.unique()

array([1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004,
       2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015,
       2016, 2017, 2018], dtype=int64)

In [12]:
#To find out if the fuel type codes
energy.fuel_type_code_pudl.unique()

array(['coal', 'gas', 'nuclear', 'oil', 'waste', 'other'], dtype=object)

In [13]:
#grouping the data by their fuel type code and sum the for each year
fuel_data=pd.DataFrame(energy.groupby(['fuel_type_code_pudl','report_year'])[['fuel_qty_burned','fuel_mmbtu_per_unit','fuel_cost_per_unit_burned','fuel_cost_per_unit_delivered','fuel_cost_per_mmbtu']].sum())
fuel_data

fuel_qty_burned  fuel_mmbtu_per_unit  \
fuel_type_code_pudl report_year                                         
coal                1994            7.172082e+08         10293.429297   
                    1995            1.257398e+09         10262.019015   
                    1996            6.988514e+08          9269.878000   
                    1997            8.169024e+08          9000.095755   
                    1998            8.982095e+08          9212.781694   
                    1999            8.805301e+08          9052.256583   
                    2000            8.620492e+08          8584.171000   
                    2001            8.564316e+08          7574.081578   
                    2002            8.144357e+08          6967.923735   
                    2003            8.018815e+08          6906.303079   
                    2004            7.735318e+08          6677.321225   
                    2005            6.046304e+08          6712.699417   
                    2006            6.002375e+08          6770.056762   
                    2007            6.131788e+08          6806.668782   
                    2008            5.752296e+08          6577.120958   
                    2009            5.219981e+08          6466.146698   
                    2010            5.931191e+08          6420.927249   
                    2011            5.228667e+08          6535.775560   
                    2012            4.833351e+08          6595.084441   
                    2013            5.001244e+08          6478.798444   
                    2014            4.571944e+08          5932.994198   
                    2015            3.878417e+08          5300.479998   
                    2016            3.804213e+08          4652.271091   
                    2017            3.297659e+08          4343.490065   
                    2018            3.216100e+08          4066.708564   
gas                 1994            2.432212e+09           498.035534   
                    1995            2.570880e+09           513.394734   
                    1996            1.915218e+09           434.236843   
                    1997            2.039925e+09           456.865725   
                    1998            2.554651e+09           526.381050   
...                                          ...                  ...   
other               2012            3.006978e+06             0.167199   
                    2013            4.387423e+06             0.152937   
                    2014            5.187094e+06             0.001414   
                    2015            5.416122e+06             0.001411   
                    2016            2.763596e+06             0.001410   
waste               1994            1.242241e+06             0.577288   
                    1995            1.642051e+06             0.337080   
                    1996            1.102632e+06             0.364899   
                    1997            1.528774e+06             0.382760   
                    1998            1.624599e+06             0.221197   
                    1999            1.108979e+06             0.186733   
                    2000            1.271108e+06             0.152991   
                    2001            1.361951e+06             0.160667   
                    2002            1.394020e+06             8.665753   
                    2003            1.082223e+06             8.818894   
                    2004            1.110020e+06             8.667558   
                    2005            1.605762e+06             8.653393   
                    2006            1.828512e+06             8.685188   
                    2007            2.153725e+06             8.676567   
                    2008            2.147651e+06             8.663653   
                    2009            2.139046e+06             8.700041   
                    2010            2.423293e+06             8.675162   
                  

In [14]:
#Further simplifying the data
new_energy=energy[["fuel_qty_burned","fuel_mmbtu_per_unit","fuel_cost_per_unit_burned","fuel_cost_per_unit_delivered","fuel_cost_per_mmbtu"]]
new_energy

fuel_qty_burned  fuel_mmbtu_per_unit  fuel_cost_per_unit_burned  \
0            5377489.0            16.590000                     18.590   
1           10486945.0            16.592000                     18.580   
2            2978683.0            24.130000                     39.720   
3            3739484.0            23.950000                     47.210   
4              40533.0             1.000000                      2.770   
5            2124933.0            23.922000                     44.240   
6               2260.0             0.064094                     28.770   
7            5372133.0            24.840000                     58.540   
8             817520.0            24.290000                     33.970   
9            1471233.0            22.484000                     28.730   
10            262720.0             1.026000                      1.940   
11            476779.0            22.428000                     28.710   
12              6673.0             1.026000                      1.940   
13           1655558.0            25.038000                     30.248   
14              6510.0             5.875338                     32.130   
15           3520831.0            24.508000                     44.404   
16           5327281.0            24.538000                     44.312   
17            666975.0            25.836000                     36.268   
18            391864.0            25.160000                     46.024   
19              3505.0             5.774958                     33.321   
20           2837040.0            24.526000                     39.911   
21             50013.0             5.845644                     30.319   
22            383043.0            24.334000                     39.850   
23             13969.0             5.742870                     32.184   
24           1338459.0            24.334000                     40.144   
25             48750.0             5.742828                     32.186   
26           2062799.0            19.970000                     31.520   
27             10088.0             5.851020                     32.650   
28             38086.0             0.880901                      2.490   
29           2599962.0            17.530000                     21.170   
...                ...                  ...                        ...   
29493          25325.0             1.033011                      6.462   
29494        2073725.0             1.038883                      4.462   
29495          10030.0             5.846053                     94.777   
29496        2461276.0             1.026999                      2.328   
29497        5853049.0             1.045818                      3.681   
29498         170018.0             1.019451                      4.536   
29499            527.0             5.816407                    102.110   
29500             36.0             5.795044                    102.488   
29501        4189920.0            16.673825                     29.799   
29502          19764.0             5.835849                     88.553   
29503        1761218.0            17.888320                     26.775   
29504          88846.0             1.031133                      6.274   
29505             81.0             5.694696                    101.330   
29506            133.0             5.756940                     95.530   
29507            320.0             5.719602                     98.180   
29508        2154971.0             1.130000                    255.000   
29509         162132.0             1.130000                      2.460   
29510         569800.0            16.110000                     14.760   
29511         613882.0             1.050000                      3.520   
29512          19655.0             1.050000                      8.910   
29513          39679.0             1.059000                      5.680   
29514            487.0             5.831070                      0.000   
29515      

### HIGHEST AVERAGE FUEL COST PER UNIT DELIVERED

In [15]:
#average fuel cost per unit delivered
fuel_avg=pd.DataFrame(energy.groupby(['report_year'])[['fuel_cost_per_unit_delivered']].mean())
fuel_avg

fuel_cost_per_unit_delivered
report_year                              
1994                            63.636060
1995                            32.735269
1996                          9196.705948
1997                         11140.197239
1998                           287.154420
1999                            25.551627
2000                           985.362877
2001                            60.050396
2002                            47.594361
2003                            55.663493
2004                           139.524275
2005                            41.438184
2006                            38.657484
2007                            43.325023
2008                            58.588197
2009                           652.694163
2010                            91.862105
2011                            59.774667
2012                            60.994502
2013                           172.307591
2014                           192.737183
2015                           326.535511
2016                           103.901761
2017                            46.196861
2018                           499.269966

##### From the above, the highest average fuel cost per unit delivered is for the year 1997

### STANDARD VARIATION AND PERCENTILE

In [16]:
#To find standard variation for fuel mmbtu per unit
standard_variation=new_energy["fuel_mmbtu_per_unit"].std()
standard_variation

10.600220307806714

In [17]:
#To find percentile for fuel mmbtu per unit
percent_ile = np.percentile(new_energy["fuel_mmbtu_per_unit"], q=[75])
percent_ile

array([17.006])

##### From the above, the standard variation to 2dp is 10.60 while the percentile to 2dp is 17.01

### SKEWNESS AND KURTOSIS

In [18]:
#To find skewness for fuel quantity burned
skewness=new_energy["fuel_qty_burned"].skew(axis = 0, skipna = True)
skewness

15.851495469109503

In [19]:
#to find kurtosis for fuel quantity burned
kurt=new_energy["fuel_qty_burned"].kurtosis()
kurt

651.3694501337732

##### From the above, the skewness and kurtosis to 2dp is 15.85 and 651.37 respectively

### MISSING VALUE PERCENTAGE

In [20]:
#Checking for missing values
energy_missing=energy.isna().sum()
energy_missing

record_id                         0
utility_id_ferc1                  0
report_year                       0
plant_name_ferc1                  0
fuel_type_code_pudl               0
fuel_unit                       180
fuel_qty_burned                   0
fuel_mmbtu_per_unit               0
fuel_cost_per_unit_burned         0
fuel_cost_per_unit_delivered      0
fuel_cost_per_mmbtu               0
dtype: int64

In [21]:
len(energy)

29523

In [22]:
percentage_missing= (energy_missing/len(energy))*100
percentage_missing

record_id                       0.000000
utility_id_ferc1                0.000000
report_year                     0.000000
plant_name_ferc1                0.000000
fuel_type_code_pudl             0.000000
fuel_unit                       0.609694
fuel_qty_burned                 0.000000
fuel_mmbtu_per_unit             0.000000
fuel_cost_per_unit_burned       0.000000
fuel_cost_per_unit_delivered    0.000000
fuel_cost_per_mmbtu             0.000000
dtype: float64

##### From the above, we can see that the percentage of missing data is 0.61% in "fuel unit" which is a categorical column. The missing values can be filled using mode computation method

### 2nd and 3rd LOWEST CORRELATION OF FUEL COST PER UNIT BURNED

In [23]:
#finding correlation
correlation = new_energy.corr()
correlation

fuel_qty_burned  fuel_mmbtu_per_unit  \
fuel_qty_burned                      1.000000            -0.080946   
fuel_mmbtu_per_unit                 -0.080946             1.000000   
fuel_cost_per_unit_burned           -0.018535            -0.010034   
fuel_cost_per_unit_delivered        -0.003551            -0.009039   
fuel_cost_per_mmbtu                 -0.001896            -0.005884   

                              fuel_cost_per_unit_burned  \
fuel_qty_burned                               -0.018535   
fuel_mmbtu_per_unit                           -0.010034   
fuel_cost_per_unit_burned                      1.000000   
fuel_cost_per_unit_delivered                   0.011007   
fuel_cost_per_mmbtu                           -0.000437   

                              fuel_cost_per_unit_delivered  \
fuel_qty_burned                                  -0.003551   
fuel_mmbtu_per_unit                              -0.009039   
fuel_cost_per_unit_burned                         0.011007   
fuel_cost_per_unit_delivered                      1.000000   
fuel_cost_per_mmbtu                              -0.000109   

                              fuel_cost_per_mmbtu  
fuel_qty_burned                         -0.001896  
fuel_mmbtu_per_unit                     -0.005884  
fuel_cost_per_unit_burned               -0.000437  
fuel_cost_per_unit_delivered            -0.000109  
fuel_cost_per_mmbtu                      1.000000

##### From the above, we can see that the 2nd and 3rd lowest correlation is fuel cost per unit delivered and fuel qty burned

### PERCENTAGE CHANGE IN FUEL COST PER UNIT BURNED FOR THE YEAR 1994 and 1998

In [24]:
#extracting values for year 1994
change=energy[energy['report_year'] == 1994]
change

record_id  utility_id_ferc1  report_year  \
0       f1_fuel_1994_12_1_0_7                 1         1994   
1      f1_fuel_1994_12_1_0_10                 1         1994   
2       f1_fuel_1994_12_2_0_1                 2         1994   
3       f1_fuel_1994_12_2_0_7                 2         1994   
4      f1_fuel_1994_12_2_0_10                 2         1994   
5      f1_fuel_1994_12_2_0_13                 2         1994   
6       f1_fuel_1994_12_2_1_1                 2         1994   
7      f1_fuel_1994_12_2_1_10                 2         1994   
8      f1_fuel_1994_12_2_2_10                 2         1994   
9       f1_fuel_1994_12_4_0_1                 4         1994   
10      f1_fuel_1994_12_4_0_2                 4         1994   
11      f1_fuel_1994_12_4_0_4                 4         1994   
12      f1_fuel_1994_12_4_0_5                 4         1994   
13      f1_fuel_1994_12_6_0_1                 6         1994   
14      f1_fuel_1994_12_6_0_2                 6         1994   
15      f1_fuel_1994_12_6_0_4                 6         1994   
16      f1_fuel_1994_12_6_0_7                 6         1994   
17     f1_fuel_1994_12_6_0_10                 6         1994   
18     f1_fuel_1994_12_6_0_13                 6         1994   
19     f1_fuel_1994_12_6_0_14                 6         1994   
20      f1_fuel_1994_12_6_1_7                 6         1994   
21      f1_fuel_1994_12_6_1_8                 6         1994   
22     f1_fuel_1994_12_6_1_10                 6         1994   
23     f1_fuel_1994_12_6_1_11                 6         1994   
24     f1_fuel_1994_12_6_1_13                 6         1994   
25     f1_fuel_1994_12_6_1_14                 6         1994   
26      f1_fuel_1994_12_7_0_1                 7         1994   
27      f1_fuel_1994_12_7_0_2                 7         1994   
28      f1_fuel_1994_12_7_0_3                 7         1994   
29      f1_fuel_1994_12_7_0_7                 7         1994   
...                       ...               ...          ...   
1205  f1_fuel_1994_12_57_3_14                57         1994   
1206   f1_fuel_1994_12_57_4_2                57         1994   
1207   f1_fuel_1994_12_41_0_1                41         1994   
1208   f1_fuel_1994_12_41_0_4                41         1994   
1209   f1_fuel_1994_12_41_0_5                41         1994   
1210   f1_fuel_1994_12_41_0_7                41         1994   
1211  f1_fuel_1994_12_41_0_10                41         1994   
1212  f1_fuel_1994_12_41_0_14                41         1994   
1213   f1_fuel_1994_12_41_1_1                41         1994   
1214  f1_fuel_1994_12_41_1_10                41         1994   
1215  f1_fuel_1994_12_41_1_11                41         1994   
1216   f1_fuel_1994_12_41_2_3                41         1994   
1217   f1_fuel_1994_12_41_2_6                41         1994   
1218  f1_fuel_1994_12_41_2_12                41         1994   
1219  f1_fuel_1994_12_41_2_15                41         1994   
1220   f1_fuel_1994_12_41_3_3                41         1994   
1221   f1_fuel_1994_12_45_0_1                45         1994   
1222   f1_fuel_1994_12_45_0_4                45         1994   
1223   f1_fuel_1994_12_45_0_7                45         1994   
1224  f1_fuel_1994_12_45_0_10                45         1994   
1225  f1_fuel_1994_12_45_0_12                45         1994   
1226  f1_fuel_1994_12_45_0_13                45         1994   
1227   f1_fuel_1994_12_45_1_1                45         1994   
1228   f1_fuel_1994_12_45_1_4                45         1994   
1229   f1_fuel_1994_12_45_1_7                45         1994   
1230  f1_fuel_1994_12_45_1_10                45         1994   
1231  f1_fuel_1994_12_45_1_13                45         1994   
1232   f1_fuel_1994_12_45_2_1                45         1994   
1233   f1_fuel_1994_12_45_2_4                45         1994   
1234   f1_fuel_1994_12_45_2_7                45         1994   

          plant_name_ferc1 fuel_type_code_pudl fuel

In [25]:
#extracting values for year 1998
change2=energy[energy['report_year'] == 1998]
change2

record_id  utility_id_ferc1  report_year  \
4618     f1_fuel_1998_12_1_0_7                 1         1998   
4619    f1_fuel_1998_12_1_0_10                 1         1998   
4620     f1_fuel_1998_12_4_0_1                 4         1998   
4621     f1_fuel_1998_12_4_0_2                 4         1998   
4622     f1_fuel_1998_12_4_0_4                 4         1998   
4623     f1_fuel_1998_12_4_0_5                 4         1998   
4624     f1_fuel_1998_12_6_0_1                 6         1998   
4625     f1_fuel_1998_12_6_0_4                 6         1998   
4626     f1_fuel_1998_12_6_0_7                 6         1998   
4627    f1_fuel_1998_12_6_0_10                 6         1998   
4628    f1_fuel_1998_12_6_0_13                 6         1998   
4629    f1_fuel_1998_12_6_0_14                 6         1998   
4630     f1_fuel_1998_12_6_1_7                 6         1998   
4631    f1_fuel_1998_12_6_1_10                 6         1998   
4632    f1_fuel_1998_12_6_1_13                 6         1998   
4633     f1_fuel_1998_12_7_0_1                 7         1998   
4634     f1_fuel_1998_12_7_0_2                 7         1998   
4635     f1_fuel_1998_12_7_0_3                 7         1998   
4636     f1_fuel_1998_12_7_0_7                 7         1998   
4637     f1_fuel_1998_12_7_0_8                 7         1998   
4638    f1_fuel_1998_12_7_0_10                 7         1998   
4639    f1_fuel_1998_12_7_0_11                 7         1998   
4640    f1_fuel_1998_12_7_0_13                 7         1998   
4641     f1_fuel_1998_12_7_1_1                 7         1998   
4642    f1_fuel_1998_12_7_1_10                 7         1998   
4643    f1_fuel_1998_12_7_1_11                 7         1998   
4644     f1_fuel_1998_12_7_2_2                 7         1998   
4645     f1_fuel_1998_12_7_2_7                 7         1998   
4646    f1_fuel_1998_12_7_2_11                 7         1998   
4647    f1_fuel_1998_12_7_2_13                 7         1998   
...                        ...               ...          ...   
5695  f1_fuel_1998_12_138_0_15               138         1998   
5696   f1_fuel_1998_12_138_1_1               138         1998   
5697   f1_fuel_1998_12_138_1_4               138         1998   
5698  f1_fuel_1998_12_138_1_10               138         1998   
5699  f1_fuel_1998_12_138_1_12               138         1998   
5700  f1_fuel_1998_12_138_1_13               138         1998   
5701   f1_fuel_1998_12_138_2_1               138         1998   
5702   f1_fuel_1998_12_145_0_1               145         1998   
5703   f1_fuel_1998_12_145_0_3               145         1998   
5704   f1_fuel_1998_12_145_0_4               145         1998   
5705   f1_fuel_1998_12_145_0_6               145         1998   
5706   f1_fuel_1998_12_145_0_7               145         1998   
5707   f1_fuel_1998_12_145_0_9               145         1998   
5708  f1_fuel_1998_12_145_0_10               145         1998   
5709  f1_fuel_1998_12_145_0_12               145         1998   
5710  f1_fuel_1998_12_145_0_13               145         1998   
5711  f1_fuel_1998_12_145_0_15               145         1998   
5712   f1_fuel_1998_12_145_1_1               145         1998   
5713   f1_fuel_1998_12_145_1_3               145         1998   
5714   f1_fuel_1998_12_145_1_4               145         1998   
5715   f1_fuel_1998_12_145_1_6               145         1998   
5716   f1_fuel_1998_12_145_1_7               145         1998   
5717   f1_fuel_1998_12_145_1_9               145         1998   
5718  f1_fuel_1998_12_145_1_12               145         1998   
5719  f1_fuel_1998_12_145_1_15               145         1998   
5720   f1_fuel_1998_12_145_2_2               145         1998   
5721   f1_fuel_1998_12_145_2_3               145         1998   
5722   f1_fuel_1998_12_145_2_6               145         1998   
5723   f1_fuel_1998_12_145_2_9               145         1998   
5724  f1_fuel_1998_12_145_2_12               145        

In [26]:
#joining the two database together
fuel_change = pd.concat([change, change2])
fuel_change

record_id  utility_id_ferc1  report_year  \
0        f1_fuel_1994_12_1_0_7                 1         1994   
1       f1_fuel_1994_12_1_0_10                 1         1994   
2        f1_fuel_1994_12_2_0_1                 2         1994   
3        f1_fuel_1994_12_2_0_7                 2         1994   
4       f1_fuel_1994_12_2_0_10                 2         1994   
5       f1_fuel_1994_12_2_0_13                 2         1994   
6        f1_fuel_1994_12_2_1_1                 2         1994   
7       f1_fuel_1994_12_2_1_10                 2         1994   
8       f1_fuel_1994_12_2_2_10                 2         1994   
9        f1_fuel_1994_12_4_0_1                 4         1994   
10       f1_fuel_1994_12_4_0_2                 4         1994   
11       f1_fuel_1994_12_4_0_4                 4         1994   
12       f1_fuel_1994_12_4_0_5                 4         1994   
13       f1_fuel_1994_12_6_0_1                 6         1994   
14       f1_fuel_1994_12_6_0_2                 6         1994   
15       f1_fuel_1994_12_6_0_4                 6         1994   
16       f1_fuel_1994_12_6_0_7                 6         1994   
17      f1_fuel_1994_12_6_0_10                 6         1994   
18      f1_fuel_1994_12_6_0_13                 6         1994   
19      f1_fuel_1994_12_6_0_14                 6         1994   
20       f1_fuel_1994_12_6_1_7                 6         1994   
21       f1_fuel_1994_12_6_1_8                 6         1994   
22      f1_fuel_1994_12_6_1_10                 6         1994   
23      f1_fuel_1994_12_6_1_11                 6         1994   
24      f1_fuel_1994_12_6_1_13                 6         1994   
25      f1_fuel_1994_12_6_1_14                 6         1994   
26       f1_fuel_1994_12_7_0_1                 7         1994   
27       f1_fuel_1994_12_7_0_2                 7         1994   
28       f1_fuel_1994_12_7_0_3                 7         1994   
29       f1_fuel_1994_12_7_0_7                 7         1994   
...                        ...               ...          ...   
5695  f1_fuel_1998_12_138_0_15               138         1998   
5696   f1_fuel_1998_12_138_1_1               138         1998   
5697   f1_fuel_1998_12_138_1_4               138         1998   
5698  f1_fuel_1998_12_138_1_10               138         1998   
5699  f1_fuel_1998_12_138_1_12               138         1998   
5700  f1_fuel_1998_12_138_1_13               138         1998   
5701   f1_fuel_1998_12_138_2_1               138         1998   
5702   f1_fuel_1998_12_145_0_1               145         1998   
5703   f1_fuel_1998_12_145_0_3               145         1998   
5704   f1_fuel_1998_12_145_0_4               145         1998   
5705   f1_fuel_1998_12_145_0_6               145         1998   
5706   f1_fuel_1998_12_145_0_7               145         1998   
5707   f1_fuel_1998_12_145_0_9               145         1998   
5708  f1_fuel_1998_12_145_0_10               145         1998   
5709  f1_fuel_1998_12_145_0_12               145         1998   
5710  f1_fuel_1998_12_145_0_13               145         1998   
5711  f1_fuel_1998_12_145_0_15               145         1998   
5712   f1_fuel_1998_12_145_1_1               145         1998   
5713   f1_fuel_1998_12_145_1_3               145         1998   
5714   f1_fuel_1998_12_145_1_4               145         1998   
5715   f1_fuel_1998_12_145_1_6               145         1998   
5716   f1_fuel_1998_12_145_1_7               145         1998   
5717   f1_fuel_1998_12_145_1_9               145         1998   
5718  f1_fuel_1998_12_145_1_12               145         1998   
5719  f1_fuel_1998_12_145_1_15               145         1998   
5720   f1_fuel_1998_12_145_2_2               145         1998   
5721   f1_fuel_1998_12_145_2_3               145         1998   
5722   f1_fuel_1998_12_145_2_6               145         1998   
5723   f1_fuel_1998_12_145_2_9               145         1998   
5724  f1_fuel_1998_12_145_2_12               145        

In [27]:
#grouping the fuel change database by report year
fuel=pd.DataFrame(fuel_change.groupby(['report_year'])[['fuel_cost_per_unit_burned']].sum())
fuel

fuel_cost_per_unit_burned
report_year                           
1994                        263292.115
1998                        140238.656

In [28]:
#finding the percent change

fuel.pct_change()

fuel_cost_per_unit_burned
report_year                           
1994                               NaN
1998                         -0.467365

#### From the above, the percent change for fuel cost per unit burned for the year 1994 and 1998 is -46.7%